src/loan/loan_request.ts

```diff
@@ -0,0 +1,324 @@
+        const collateral = new TokenAmount(collateralAmount, collateralToken);
+        const interestRateTyped = new InterestRate(interestRate);
+        const termLength = new TimeInterval(termDuration, termUnit);
+        const debtorAddressTyped = new EthereumAddress(debtorAddress);
+        const expiresIn = new TimeInterval(expiresInDuration, expiresInUnit);
+
+        const currentBlocktime = new BigNumber(await dharma.blockchain.getCurrentBlockTime());
+
+        const expirationTimestampInSec = expiresIn.fromTimestamp(currentBlocktime);
+
+        const loanRequestConstructorParams: BaseLoanConstructorParams = {
+            principal,
+            collateral,
+            interestRate: interestRateTyped,
+            termLength,
+            debtorAddress: debtorAddressTyped,
+            expiresAt: expirationTimestampInSec.toNumber(),
+        };
+
+        const loanOrder: CollateralizedSimpleInterestLoanOrder = {
+            principalAmount: principal.rawAmount,
+            principalTokenSymbol: principal.tokenSymbol,
+            interestRate: interestRateTyped.raw,
+            amortizationUnit: termLength.getAmortizationUnit(),
+            termLength: new BigNumber(termLength.amount),
+            collateralTokenSymbol: collateral.tokenSymbol,
+            collateralAmount: collateral.rawAmount,
+            gracePeriodInDays: new BigNumber(0),
+            expirationTimestampInSec,
+        };
+
+        const data = await dharma.adapters.collateralizedSimpleInterestLoan.toDebtOrder(loanOrder);
+        const debtKernel = await dharma.contracts.loadDebtKernelAsync();
+        const repaymentRouter = await dharma.contracts.loadRepaymentRouterAsync();
+        const salt = this.generateSalt();
+
+        data.debtor = debtorAddressTyped.toString();
+        data.kernelVersion = debtKernel.address;
+        data.issuanceVersion = repaymentRouter.address;
+        data.salt = salt;
+
+        const loanRequest = new LoanRequest(dharma, loanRequestConstructorParams, data);
+
+        await loanRequest.signAsDebtor();
+
+        return loanRequest;
+    }
+
+    public static async load(dharma: Dharma, data: DebtOrderData): Promise<LoanRequest> {
+        const loanOrder = await dharma.adapters.collateralizedSimpleInterestLoan.fromDebtOrder(
+            data,
+        );
+
+        const principal = TokenAmount.fromRaw(
+            loanOrder.principalAmount,
+            loanOrder.principalTokenSymbol,
+        );
+
+        const collateral = TokenAmount.fromRaw(
+            loanOrder.collateralAmount,
+            loanOrder.collateralTokenSymbol,
+        );
+
+        const interestRate = InterestRate.fromRaw(loanOrder.interestRate);
+
+        const termLength = new TimeInterval(
+            loanOrder.termLength.toNumber(),
+            loanOrder.amortizationUnit,
+        );
+
+        const debtorAddress = new EthereumAddress(loanOrder.debtor!); // TODO(kayvon): this could throw.
```

Comment:
What needs to be done for this TODO?

Reply:
loan orders shouldn't hold onto optional properties. we can address this down the line.